### ISSUE TITLE
Text input character delay

### ISSUE THREAD

**Issue by adam-mcdaniel** [fschutt/azul#202]
## Description

In the text_input example, the text input is delayed by one character. For example, if I type `hey jude`, `hey jud` is displayed until I hit another key.

### Operating system & Rust
Pop!OS 19.04 using Rust version 1.36.0

### Windowing System

Im using X11.

## Steps to Reproduce

Here is my adapted version of the `text_input` example.

```rust
extern crate azul;
use azul::prelude::*;
use azul::widgets::text_input::*;


// extern crate rusty_yaml;
// use rusty_yaml::*;

// mod buildbot;
// use buildbot::{MasterYaml};




/// This will be used to run Rusty-CI commands like `Setup`,
/// `Build`, `Start`, `Kill`, and `Install`
struct RustyCIGui {
    text_input: TextInputState,
    // master_yaml: MasterYaml
}


impl Default for RustyCIGui {
    fn default() -> Self {
        Self {
            text_input: TextInputState::new("Hover mouse over rectangle and press keys"),
        }
    }
}

impl Layout for RustyCIGui {
    fn layout(&self, info: LayoutInfo<Self>) -> Dom<Self> {
        TextInput::new()
            .bind(info.window, &self.text_input, &self)
            .dom(&self.text_input)
            .with_id("text_input_1")
    }
}



const CSS: &str = "
#text_input_1 {
    color: red;
    font-color: blue;
    border-radius: 2px;
    width: 1000px;
    height: 100px;
    border: 2px solid transparent;
    background-color: #ccc;
    position: absolute;
    top: 100px;
    left: 100px;
}

#text_input_1 p {
    font-size: 20px;
}
#text_input_1:focus {
    border: 10px solid  #80ff80;
}
";

fn main() {
    let mut app = App::new(
        RustyCIGui::default(),
        AppConfig::default()
    ).unwrap();

    let css = css::override_native(CSS).unwrap();
    
    let window = app.create_window(
        WindowCreateOptions::default(), css
    ).unwrap();

    app.run(window).unwrap();
}
```

**Comment by fschutt** [comment]
Yes, I know about this bug, filing this under "redrawing bugs". Again, this should be fixed with winits "event loop 2.0" system.

**Comment by fschutt** [comment]
This has been fixed, the display is automatically updated after the character is pressed. Azul now contains an optimization for only exchanging strings on the screen (one of the most common things when typing), so that the layout doesn't have to be completely recalculated.

winit has been dropped, it's almost impossible to get anything done with winit and not run into redrawing bugs. I will have to write code for each platform specifically, instead of using glutin / winit.
